packages/flutter/android/build.gradle

```diff
@@ -52,8 +56,22 @@ android {
         sourceCompatibility JavaVersion.VERSION_1_8
         targetCompatibility JavaVersion.VERSION_1_8
     }
+    
+    kotlinOptions {
+        jvmTarget = '1.8'
+    }
+
+    sourceSets {
+        main.java.srcDirs += 'src/main/kotlin'
+    }
 
     defaultConfig {
-        minSdkVersion 16
+        minSdkVersion 24
     }
 }
+
+dependencies {
+    implementation "net.java.dev.jna:jna:5.14.0@aar"
+    /* JSON serialization */
+    implementation 'org.jetbrains.kotlinx:kotlinx-serialization-json:1.6.3'
```

Comment:
Kotlin 1.8.20+ has serialization plugin. I don't think we need to depend on it explicitly.

Reply:
Doesn't compile without this